pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -287,6 +294,10 @@ func (d *DualWriterMode2) Update(ctx context.Context, name string, objInfo rest.
 	log := d.Log.WithValues("name", name, "method", method)
 	ctx = klog.NewContext(ctx, log)
 
+	// The incoming RV is not stable -- it may be from legacy or storage!
+	// This sets a flag in the context and our apistore is more lenient when it exists
+	ctx = context.WithValue(ctx, dualWriteContextKey{}, true)
```

Comment:
should this be unset when writing to the new storage? also, it seems like it would need to exist in modes 1 & 3 as well 🤔

Reply:
> should this be unset when writing to the new storage?

If we *know* the input RV+UID is from storage then yes, but as I read the code it seems that in mode 2 it could be either.  So just depends if we want to fail or not 🤷🏻 

Good point on mode1+3 -- things "work" because we silently fail the async updates to the oposite storage.
https://github.com/grafana/grafana/blob/main/pkg/apiserver/rest/dualwriter_mode1.go#L291
https://github.com/grafana/grafana/blob/main/pkg/apiserver/rest/dualwriter_mode3.go#L247

I'll add the flag and see if tests still pass!